static/js/stream_list.js

```diff
@@ -488,6 +488,10 @@ exports.set_event_handlers = function () {
         const sub = stream_data.get_sub_by_id(stream_id);
         popovers.hide_all();
         narrow.by("stream", sub.name, {trigger: "sidebar"});
+        if ($("#gear-menu").hasClass("open")) {
+            $("#gear-menu").removeClass("open");
+            $("#gear-menu").addClass("active");
+        }
```

Comment:
If someone presses on any stream in the left sidebar , the setting dropdown must collapse ( If it is open ) .

Reply:
We could probably extract these three lines to a helper called `gear_menu.close_if_needed()`.  The module `gear_menu.js` already exists.  And then we can call that helper from both places.